### DISCUSSION TITLE
[Feature Request] SetGame without reseting the current game

### DISCUSSION THREAD

**Discussion by DarkBahamut162** [TeamRizu/OutFox#519] [03 - Feature Requests]
### Is your feature request related to a problem?

I can't continue on with my theme with it behaving like this.
I would like to insert "ScreenSelectGameMode" into the game along with "ScreenSelectPlayMode" and "ScreenSelectStyle" and "ScreenSelectProfile" (in that order) while also behaving like them within the same theme.
It currently resets everything (as well as needed variables needed for animation and functionality).

### Describe the solution you'd like

There is the following code within `GameLoop` inside the function `DoChangeGame()`:
```
if(theme_changing){
  SAFE_DELETE(SCREENMAN);
  TEXTUREMAN->DoDelayedDelete();
  LUA->RegisterTypes();
  THEME->SwitchThemeAndLanguage(g_NewTheme, THEME->GetCurLanguage(),PREFSMAN->m_bPseudoLocalize);
  PREFSMAN->m_sTheme.Set(g_NewTheme);
  StepMania::ApplyGraphicOptions();
  SCREENMAN= new ScreenManager();
}
StepMania::ResetGame();
```
The `StepMania::ResetGame();` should be inside the upper code for my stuff to work fine (from what I can see)

### Describe alternatives you've considered

I'm just looking for something like what "ScreenSelectPlayMode" and "ScreenSelectStyle" are doing.
I don't think it is actually necessary to reset the entire game if it all stays within the same theme.

### Additional context

Not sure if this counts as a "Unexpected Behavior", "Nitpick" or even a "Bug"

**Comment by Jousway** [comment]
you can already change gamemode in an theme without switching the theme since SM4, just use `GAMEMAN:SetGame('dance','themename')`

**Comment by DarkBahamut162** [comment]
I already did that and its both the gamemode AND the theme I'm already in but it STILL resets the entire game as well as ALL variables.
I even did the following afterwards:

`SCREENMAN:GetTopScreen():SetNextScreenName("ScreenSelectPlayMode"):StartTransitioningScreen("SM_GoToNextScreen")`

...and it STILL resets everything and puts me to `ScreenInit`
And even with the following code in the metrics:
```
[Common]
...
AfterGameChangeScreen="ScreenSelectPlayMode"
```
...it then gets me towards said screen I wanted to reach afterwards but all variables are reset which is a bad thing for me.

My problem is the `StepMania::ResetGame();` OUTSIDE of the `if(theme_changing){...}`
Move it inside and I'm happy.
